msticpy/nbtools/timeseries.py

```diff
@@ -242,3 +244,76 @@ def display_timeseries_anomolies(
 
     show(plot)
     return plot
+
+
+@export
+def timeseries_anomalies_stl(data: pd.DataFrame, **kwargs,) -> pd.DataFrame:
+    """ Discover anomalies in Timeseries data using
+    STL (Seasonal-Trend Decomposition using LOESS) method using statsmodels package.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        DataFrame as a time series data set retrived from data connector or external data source.
+        Dataframe must have 2 columns with time column set as index and other numeric value.
+
+    Other Parameters
+    ----------------
+    seasonal : int, optional
+        Seasonality period of the input data required for STL. 
+        Must be an odd integer, and should normally be >= 7 (default).
+    period: int, optional
+        Periodicity of the the input data. by default 24 (Hourly).
+    score_threshold : float, optional
+        standard deviation threshold value calculated using Z-score used to flag anomalies,
+        by default 3
+
+    Returns
+    -------
+    pd.DataFrame
+        Returns a dataframe with additional columns by decomposing time series data
+        into residual, trend, seasonal, weights, baseline, score and anomalies.
+        The anomalies column will have 0, 1,-1 values based on score_threshold set.
+    """
+    seasonal: int = kwargs.get("seasonal", 7)
```

Comment:
You might want to use the common.utils.check_kwargs() function here. It checks for bad arguments and will suggest closest matches (e.g. if someone sends threshold=20 it will tell them that the closest match is score_threshold)

Reply:
thanks. added accordingly.